src/main/java/Duke.java

```diff
@@ -1,10 +1,174 @@
+    private static final String MESSAGE_LIST_HEADER = "Here are the tasks in your list:";
+    private static final String MESSAGE_MARK = "Nice! I've marked this task as done:";
+    private static final String MESSAGE_UNMARK = "OK, I've marked this task as not done yet:";
+    private static final String MESSAGE_WELCOME = "Hello! I'm FredBot.\nWhat can I do for you?";
+
+    private static final String PREFIX_BY = "/by";
+    private static final String PREFIX_FROM = "/from";
+    private static final String PREFIX_TO = "/to";
+
+    private static final Scanner SCANNER = new Scanner(System.in);
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+        initTaskList();
+        showWelcomeMessage();
+        while (true) {
+            String command = readUserInput();
+            executeCommand(command);
+        }
+    }
+
+    private static void initTaskList() {
+        allTasks = new Task[TASK_CAPACITY];
+        count = 0;
+    }
+
+    private static void showWelcomeMessage() {
+        System.out.println(MESSAGE_WELCOME);
+    }
+
+    private static String readUserInput() {
+        return SCANNER.nextLine();
+    }
+
+    private static void executeCommand(String input) {
+        final String[] commandWordAndArgs = splitCommandWordAndArgs(input);
+        final String commandWord = commandWordAndArgs[0];
+        final String commandArgs = commandWordAndArgs[1];
+
+        switch (commandWord) {
+            case COMMAND_TODO:
+                executeAddTodo(commandArgs);
+                break;
+            case COMMAND_DEADLINE:
+                executeAddDeadline(commandArgs);
+                break;
+            case COMMAND_EVENT:
+                executeAddEvent(commandArgs);
+                break;
+            case COMMAND_LIST:
+                executeList();
+                break;
+            case COMMAND_MARK:
+                executeMark(commandArgs);
+                break;
+            case COMMAND_UNMARK:
+                executeUnmark(commandArgs);
+                break;
+            case COMMAND_EXIT:
+                executeExit();
+                break;
+            default:
+                executeError();
+                break;
+        }
+    }
+
+    private static String[] splitCommandWordAndArgs(String input) {
+        final String[] split = input.split(" ", 2);
+        return split.length == 2 ? split : new String[]{split[0], ""};
+    }
+
+    private static void executeAddTodo(String input) {
+        allTasks[count] = new Todo(input);
+        echoTask();
+    }
+
+    private static void executeAddDeadline(String input) {
+        String[] split = splitDeadlineAndDate(input);
```

Comment:
I'm not sure if I like the variable name "split" for naming the string array. 
First of all, the name is not in plural form, which is not consistent with the way you name string array types (see line 4), and also obfuscates the idea that this is a collection of objects, which makes the code less readable.
Secondly, the name "split" is a verb, not a noun, which makes it hard to understand when it is a variable name. Perhaps you could rename it as "parsedArguments" to capture the idea that this string array stores the separated values of the user input